pkg/util/diagnostics/profileset.go

```diff
@@ -0,0 +1,132 @@
+package diagnostics
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	_ "net/http/pprof"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/encoding"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/rpcbus"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	errAlreadyStarted = errors.New("profile already exits")
+)
+
+// ProfileSet allows fetching program samples of different types.
+// It could be handful if node user observes problematic situation or any type of inefficiency.
+// In most cases, it will allow a developer to catch a perf issue (Memory, CPU or concurrency inefficiency) at development phase.
+// As support for periodical sample fetching is added, the tool could enable 'continuous profiling' with negligible overhead.
+//
+// See also func profile.loop() for a few examples of use
+//
+type ProfileSet struct {
+	profiles map[string]profile
+	quit     chan struct{}
+}
+
+// NewProfileSet creates and starts ProfileSet from a set of settings strings
+func NewProfileSet(profiles []string) (ProfileSet, error) {
```

Comment:
`NewProfileSet` should accept a slice of `Profile`

Reply:
Makes sense. 
It seems the config.Loader() should be capable to parse the config.
Also, grpc call with a set of fields instead of a single config string is more convenient.